mlir/lib/Transform/AIRDmaToChannel.cpp

```diff
@@ -112,106 +112,6 @@ static SmallVector<Value, 1> getLoopTokens(scf::ParallelOp loop) {
   return output;
 }
 
-static void replaceAllUsesOfInductionVarsWith(scf::ForOp old_loop,
-                                              scf::ForOp new_loop) {
-  replaceAllUsesInRegionWith(old_loop.getInductionVar(),
-                             new_loop.getInductionVar(), new_loop.getRegion());
-}
-
-static void replaceAllUsesOfInductionVarsWith(scf::ParallelOp old_loop,
-                                              scf::ParallelOp new_loop) {
-  for (unsigned i = 0; i < old_loop.getInductionVars().size(); i++) {
-    replaceAllUsesInRegionWith(old_loop.getInductionVars()[i],
-                               new_loop.getInductionVars()[i],
-                               new_loop.getRegion());
-  }
-}
-
-static void replaceAllUsesOfLoopTokensWith(scf::ForOp old_loop,
-                                           scf::ForOp new_loop) {
-  if (old_loop.getRegionIterArgs().size()) {
-    for (unsigned i = 0; i < old_loop.getRegionIterArgs().size(); i++) {
-      replaceAllUsesInRegionWith(old_loop.getRegionIterArgs()[i],
-                                 new_loop.getRegionIterArgs()[i],
-                                 new_loop.getRegion());
-    }
-  }
-}
-
-static void replaceAllUsesOfLoopTokensWith(scf::ParallelOp old_loop,
-                                           scf::ParallelOp new_loop) {
-  if (old_loop.getInitVals().size()) {
-    for (unsigned i = 0; i < old_loop.getInitVals().size(); i++) {
-      replaceAllUsesInRegionWith(old_loop.getInitVals()[i],
-                                 new_loop.getInitVals()[i],
-                                 new_loop.getRegion());
-    }
-  }
-}
-
-static void
-replaceAllUsesOfIncomingTokensWith(SmallVector<Value, 4> incoming_tokens,
-                                   scf::ForOp new_loop) {
-  for (Value v : incoming_tokens) {
-    replaceAllUsesInRegionWith(v, new_loop.getRegionIterArgs()[0],
-                               new_loop.getRegion());
-  }
-}
-
-static void
-replaceAllUsesOfIncomingTokensWith(SmallVector<Value, 4> incoming_tokens,
-                                   scf::ParallelOp new_loop) {
-  for (Value v : incoming_tokens) {
-    replaceAllUsesInRegionWith(v, new_loop.getInitVals()[0],
-                               new_loop.getRegion());
-  }
-}
-
-static void
-replaceAllUsesOfConstsInRegionWithNew(SmallVector<Value, 4> constants,
-                                      OpBuilder builder, Region &region) {
-  for (auto c : constants) {
-    replaceAllUsesInRegionWith(
-        c, builder.clone(*c.getDefiningOp())->getResult(0), region);
-  }
-}
```

Comment:
I didn't really find anywhere in any tests where these were necessary. Won't all these get captured in the backward slice? And already be remapped?

Reply:
Agreed. Those were indeed reinventing backward slice + remap. Makes sense to remove them.